blockdag/sync_rate.go

```diff
@@ -0,0 +1,52 @@
+package blockdag
+
+import "time"
+
+const syncRateWindowDuration = 15 * time.Minute
+
+// addBlockProcessTime adds the last the last
+// block process time in order to measure the
+// recent sync rate.
+//
+// This function MUST be called with the DAG state lock held (for writes).
+func (dag *BlockDAG) addBlockProcessTime() {
```

Comment:
I'm thinking how to prevent confusion between "the time the block was processed" vs "the time it took the block to process".

Not sure I have a very satisfying solution for that. Some thoughts: 
1. Use "addBlockProcessDate". This will work on clarifying the situation, but a bit silly because we are talking about 15 minute resolutions.
2. Maybe addBlockProcessTimestamp?
3. Maybe something like logBlockProcessed (but without the word "log" to prevent confusion with actual logging - think about some synonym)

Think about it and discuss with team